lib/purchases/ui/pages/main_page/ticket_card.dart

```diff
@@ -0,0 +1,52 @@
+import 'package:flutter/material.dart';
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+import 'package:myecl/purchases/class/ticket.dart';
+import 'package:myecl/tools/functions.dart';
+import 'package:myecl/tools/ui/layouts/card_layout.dart';
+
+class TicketCard extends HookConsumerWidget {
+  const TicketCard({
+    super.key,
+    required this.ticket,
+    required this.onClicked,
+  });
+
+  final Ticket ticket;
+  final VoidCallback onClicked;
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    return Padding(
+      padding: const EdgeInsets.symmetric(horizontal: 10, vertical: 5),
+      child: GestureDetector(
+        onTap: ticket.scanLeft > 0 ? onClicked : null,
+        child: CardLayout(
+          color: ticket.scanLeft > 0 ? Colors.white : Colors.grey[400]!,
+          width: double.infinity,
+          child: Padding(
+            padding: const EdgeInsets.symmetric(horizontal: 10),
+            child: Column(
+              crossAxisAlignment: CrossAxisAlignment.start,
+              children: [
+                Text(
+                  ticket.name,
+                  style: const TextStyle(
+                    fontSize: 16,
+                    fontWeight: FontWeight.bold,
+                  ),
+                ),
+                Text(
+                  "${ticket.scanLeft} scan${ticket.scanLeft > 1 ? 's' : ""} restant${ticket.scanLeft > 1 ? 's' : ""} - Valide jusqu'au ${processDate(ticket.expirationDate)}",
```

Comment:
We should use constants or a function rather than hard code text

Reply:
True, forgot to change it